### PR TITLE
MODUSERS-369 : Update to Java 17 mod-users

### PR DESCRIPTION
**Pull request by nirmalr-shah**
_[MODUSERS-369](https://issues.folio.org/browse/MODUSERS-369) : Update to Java 17 mod-users_

## Purpose
 Update the module to Java 17 and identify issues.
 https://wiki.folio.org/display/TC/JDK+17+and+Java+17

## Approach
 Changed the java sdk version to 17 wherever applicable
 
 ## Evidence of passing all Test cases in local environ
 
image

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,9 @@
-FROM folioci/alpine-jre-openjdk11:latest
+FROM folioci/alpine-jre-openjdk17:latest
+
+# Install latest patch versions of packages: https://pythonspeed.com/articles/security-updates-in-docker/
```
Comment (marcjohnson-kint) [inline Dockerfile:3]: Is this part of the documented steps for upgrading to JDK 17 or something additional included in this work?

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,9 @@
...
+USER root
+RUN apk upgrade --no-cache
+USER folio
```
Comment (marcjohnson-kint) [inline Dockerfile:6]: Is this part of the documented steps for upgrading to JDK 17 or something additional included in this work?

Comment (nirmalr-shah) [reply Dockerfile:6]: yes @marcjohnson-kint PFB the link of the same which I followed for the **_java 17 upgradation documentation_**

Link: https://wiki.folio.org/display/TC/JDK+17+and+Java+17

image

Comment (marcjohnson-kint) [reply Dockerfile:6]: @nirmalr-shah Thanks for confirming, it seems odd to me that's included in these upgrade instructions.

I'll resolve this

**Review by Vignesh-kalyanasundaram — APPROVED** [review]

**Review by gurleenkaurbp — APPROVED** [review]

**Pull request merged**